tests/ThreeWayMergeTest.php

```diff
@@ -0,0 +1,512 @@
+            I do not like doing it
+            it was easy'],
+        ];
+
+        $expected = [
+            'keyA' => ['This is fun
+            I do not like doing it
+            it was easy'],
+        ];
+        $merge = new ThreeWayMerge();
+        $result = $merge->performMerge($original, $local, $remote);
+        $this->assertEquals($expected, $result);
+    }
+    
+    public function testMultilineAdditionModification()
+    {
+        $original = [
+            'keyA' => ['This is not fun
+            I like doing it'],
+        ];
+
+        $local = [
+            'keyA' => ['This is fun
+            I do not like doing it'],
+        ];
+
+        $remote = [
+            'keyA' => ['This is not fun
+            I do not like doing it
+            it was easy'],
+        ];
+
+        $expected = [
+            'keyA' => ['This is fun
+            I do not like doing it
+            it was easy'],
+        ];
+        $merge = new ThreeWayMerge();
+        $result = $merge->performMerge($original, $local, $remote);
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMultilineRemovalModification()
+    {
+        $original = [
+            'keyA' => ['This is not fun
+            I like doing it
+            it was not easy'],
+        ];
+
+        $local = [
+            'keyA' => ['This is fun
+            I do not like doing it
+            it was not easy'],
+        ];
+
+        $remote = [
+            'keyA' => ['This is not fun
+            I like doing it'],
+        ];
+
+        $expected = [
+            'keyA' => ['This is fun
+            I do not like doing it'],
+        ];
+        $merge = new ThreeWayMerge();
+        $result = $merge->performMerge($original, $local, $remote);
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMultilineAdditionRemovalModificationTest1()
+    {
+        $original = [
+            'keyA' => ['This is not fun
+            I like doing it'],
+        ];
+
+        $local = [
+            'keyA' => ['This is fun
+            I like doing it
+            it was easy'],
+        ];
+
+        $remote = [
+            'keyA' => ['This is not fun'],
+        ];
+
+        $expected = [
+            'keyA' => ['This is fun
+            it was easy'],
+        ];
+        $merge = new ThreeWayMerge();
+        $result = $merge->performMerge($original, $local, $remote);
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testMultilineAdditionRemovalModificationTest2()
+    {
+        $original = [
+            'keyA' => ['This is not fun
+            I like doing it
+            it was easy'],
+        ];
+
+        $local = [
+            'keyA' => ['This is fun
+            I like doing it'],
+        ];
+
+        $remote = [
+            'keyA' => ['This is not fun'],
+        ];
+
+        $expected = [
+            'keyA' => ['This is fun
+            it was easy'],
+        ];
```

Comment:
Why do we expect `it was easy` here? Too me it seems like both the local and remote versions have removed this line.